### PR TITLE
Show a message when building as part of dotnet-run

### PR DESCRIPTION
**Pull request by pranavkm**
* Tweak ILaunchSettingsProvider to return a LaunchSettingsResult
* Read LaunchSettingsResult during dotnet-run to show message

Based on https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1154478

### CODE REVIEW

**Review by vijayrkn — APPROVED** [review]

File: src/Cli/dotnet/commands/dotnet-run/RunCommand.cs [hunk 1]
```diff
@@ -41,17 +39,37 @@ public int Execute()
...
-                if (!ApplyLaunchProfileSettingsIfNeeded(ref targetCommand))
+                if (launchSettings != null)
                 {
-                    return 1;
+                    if (!string.IsNullOrEmpty(launchSettings.ApplicationUrl))
+                    {
+                        targetCommand.EnvironmentVariable("ASPNETCORE_URLS", launchSettings.ApplicationUrl);
+                    }
+
+                    foreach (var entry in launchSettings.EnvironmentVariables)
```
Comment (wli3) [inline src/Cli/dotnet/commands/dotnet-run/RunCommand.cs:67]: no test coverage for this code

File: src/Cli/dotnet/commands/dotnet-run/RunCommand.cs [hunk 2]
```diff
@@ -41,17 +39,37 @@ public int Execute()
...
             try
             {
                 ICommand targetCommand = GetTargetCommand();
-                if (!ApplyLaunchProfileSettingsIfNeeded(ref targetCommand))
+                if (launchSettings != null)
                 {
-                    return 1;
+                    if (!string.IsNullOrEmpty(launchSettings.ApplicationUrl))
+                    {
+                        targetCommand.EnvironmentVariable("ASPNETCORE_URLS", launchSettings.ApplicationUrl);
```
Comment (wli3) [inline src/Cli/dotnet/commands/dotnet-run/RunCommand.cs:64]: no test coverage for this code

**Review by wli3 — CHANGES_REQUESTED** [review]
My forces is on RunCommand.cs. There are some code lacking coverage. And it is better to have unit tests to cover these code.

**Comment by pranavkm** [comment]
@wli3 I wrote a functional test for the scenarios you listed to make sure previously untested code worked after I basically moved the code around. While I understand it would be great to add unit test coverage for this type, this is not the PR to do it. I'll file an issue to add additional test coverage in this area.

**Comment by pranavkm** [comment]
Filed https://github.com/dotnet/sdk/issues/12609 as follow up.

**Comment by wli3** [comment]
> While I understand it would be great to add unit test coverage for this type, this is not the PR to do it. I'll file an issue to add additional test coverage in this area.

I don't like this approach. And you dismissed my block directly. Unless it is really hard. Tests should go with the PR.

**Pull request merged**